website/content/docs/enterprise/pkcs11-provider/aws-xks.mdx

```diff
@@ -0,0 +1,201 @@
+---
+layout: docs
+page_title: AWS KMS External Key Store (XKS) - PKCS#11 Provider - Vault Enterprise
+description: |-
+  The Vault PKCS#11 Provider can be used with AWS KMS External Key Store.
+---
+
+# Vault with AWS KMS External Key Store (XKS) via PKCS#11 and XKS Proxy
+
+~> **Important**: AWS [`xks-proxy`](https://github.com/aws-samples/aws-kms-xks-proxy) is not considered production-ready.
+
+Vault's KMIP Secrets Engine can be used as an external key store for the AWS KMS [External Key Store (XKS)](https://aws.amazon.com/blogs/aws/announcing-aws-kms-external-key-store-xks/) protocol using the AWS [`xks-proxy`](https://github.com/aws-samples/aws-kms-xks-proxy) along
+with the [Vault PKCS#11 Provider](https://developer.hashicorp.com/vault/docs/enterprise/pkcs11-provider).
+
+## Overview
+
+This is tested as working with Vault 1.11.0 Enterprise (and later) with Advanced Data Protection (KMIP support).
+
+Prerequisites:
+
+* A server capable of running XKS Proxy, which is exposed to the Internet or a VPC endpoint. This can be the same as the Vault server.
+* `libvault-pkcs11.so` downloaded from [releases.hashicorp.com](https://releases.hashicorp.com/vault-pkcs11-provider) for your platform and available on the XKS Proxy server.
+* Vault Enterprise with the KMIP Secrets Engine available and with TCP port 5696 accessible to where XKS Proxy will be running.
+
+There are 3 parts to this setup:
+
+1. Vault KMIP Secrets Engine standard setup. (There is nothing specific to XKS in this setup.)
+1. Vault PKCS#11 setup to tell the PKCS#11 provider (`libvault-pkcs11.so`) how to talk to the Vault KMIP Secrets Engine. (There is nothing specific to XKS in this setup.)
+1. XKS Proxy setup.
+
+## Vault Setup
+
+On the Vault server, we need to [setup the KMIP Secrets Engine](/docs/secrets/kmip):
+
+1. Start the [KMIP Secrets Engine](/docs/secrets/kmip) and listener:
+
+    ```sh
+    vault secrets enable kmip
+    vault write kmip/config listen_addrs=0.0.0.0:5696
+    ```
+
+    ~> **Important**: When configuring KMIP for Oracle, you will probably need to set the
+    `server_hostnames` and `server_ips` [configuration parameters](/api-docs/secret/kmip#parameters),
+    otherwise the TLS connection to the KMIP Secrets Engine will fail due to certification validation errors.
+    When configuring Oracle TDE, this error can manifest as the `sqlplus` session silently hanging.
```

Comment:
Is this oracle info relevant for XKS?

Reply:
Ah, good catch. I thought I had deleted all of the Oracle stuff from this doc again. :)